flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java

```diff
@@ -715,7 +715,12 @@ public OP getMainOperator() {
                             taskEnvironment.getUserCodeClassLoader().asClassLoader());
         }
 
-        return new RecordWriterOutput<>(recordWriter, outSerializer, sideOutputTag, this);
+        return new RecordWriterOutput<>(
+                recordWriter,
+                outSerializer,
+                sideOutputTag,
+                this,
+                edge.getPartitioner().isPointwise());
```

Comment:
Can downstream task have two different incoming edges, one pointwise the other not? If so that might require a bit code/testing (combination of non timeoutable CB and UC CB probably doesn't work) .

Reply:
I think you are right. I was falsely assuming that we decide persisting on gate level but in reality any gate can cause a `prepareSnapshot` even on the pointwise connection. 
That would effectively mean that we need to disable it on all inputs of task X if X has at least one pointwise, right?